internal/provider/resource_docker_image_funcs.go

```diff
@@ -319,6 +323,24 @@ func buildDockerImage(ctx context.Context, rawBuild map[string]interface{}, imag
 	buildOptions.Labels = labels
 	log.Printf("[DEBUG] Labels: %v\n", labels)
 
+	dockerClientVersion := client.ClientVersion()
+	log.Printf("[DEBUG] DockerClientVersion: %v, minBuildKitDockerVersion: %v\n", dockerClientVersion, minBuildkitDockerVersion)
+
+	if versions.GreaterThanOrEqualTo(dockerClientVersion, minBuildkitDockerVersion) {
+		// docker client supports BuildKit
+		log.Printf("[DEBUG] Enabling BuildKit")
+		s, _ := session.NewSession(ctx, "docker-provider", "")
+		dialSession := func(ctx context.Context, proto string, meta map[string][]string) (net.Conn, error) {
```

Comment:
Return an error or disable BuildKit.
You don't have to use closure.

Reply:
How would I write it without a closure? `s.Run()` (3 lines below), accepts a `type Dialer func(ctx context.Context, proto string, meta map[string][]string) (net.Conn, error)` as second parameter

Reply:
Oh, sorry. I misunderstood this code as the function call. 🙇‍♂️ 


```go
dialSession := func() {

}() // call the closure
```